library/spdm_responder_lib/libspdm_rsp_csr.c

```diff
@@ -162,17 +164,61 @@ libspdm_return_t libspdm_get_response_csr(libspdm_context_t *spdm_context,
 
     csr_len = *response_size - sizeof(spdm_csr_response_t);
     csr_p = (uint8_t*)(spdm_response + 1);
-    result = libspdm_gen_csr(spdm_context->connection_info.algorithm.base_hash_algo,
-                             spdm_context->connection_info.algorithm.base_asym_algo,
-                             &need_reset, request, request_size,
-                             requester_info, requester_info_length,
-                             opaque_data, opaque_data_length,
-                             &csr_len, csr_p, is_device_cert_model);
+
+    csr_tracking_tag = 0;
+    if (libspdm_get_connection_version(spdm_context) >= SPDM_MESSAGE_VERSION_13) {
+        csr_tracking_tag =
+            (spdm_request->header.param2 & SPDM_GET_CSR_REQUEST_ATTRIBUTES_CSR_TRACKING_TAG_MASK) >>
+            SPDM_GET_CSR_REQUEST_ATTRIBUTES_CSR_TRACKING_TAG_OFFSET;
+
+        if (csr_tracking_tag > SPDM_MAX_CSR_TRACKING_TAG) {
+            return libspdm_generate_error_response(spdm_context,
+                                                   SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                                   response_size, response);
+        }
+
+        if ((spdm_request->header.param2 & SPDM_GET_CSR_REQUEST_ATTRIBUTES_OVERWRITE) != 0) {
+            overwrite = true;
+        } else {
+            overwrite = false;
+        }
+        result = libspdm_gen_csr_ex(spdm_context->connection_info.algorithm.base_hash_algo,
+                                    spdm_context->connection_info.algorithm.base_asym_algo,
+                                    &need_reset, request, request_size,
+                                    requester_info, requester_info_length,
+                                    opaque_data, opaque_data_length,
+                                    &csr_len, csr_p, is_device_cert_model,
+                                    &csr_tracking_tag, overwrite);
+    } else {
+        result = libspdm_gen_csr(spdm_context->connection_info.algorithm.base_hash_algo,
+                                 spdm_context->connection_info.algorithm.base_asym_algo,
+                                 &need_reset, request, request_size,
+                                 requester_info, requester_info_length,
+                                 opaque_data, opaque_data_length,
+                                 &csr_len, csr_p, is_device_cert_model);
+    }
+
     if (!result) {
-        return libspdm_generate_error_response(
-            spdm_context,
-            SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
-            response_size, response);
+        if ((csr_tracking_tag == 0) &&
```

Comment:
The Integrator needs a better way to tell libspdm that it has run out of `CSRTrackingTag`s. Maybe have `csr_tracking_tag` return `0xFF`.

Reply:
I have made the `csr_tracking_tag` return 0xFF when the device is busy.